src/indexes/common.ts

```diff
@@ -6,10 +6,15 @@ export type TargetProperty = string;
 export type IndexOptions<T extends IndexableObj> = {
   targetProperties: Properties;
 };
-export type Extension = {
+export type Extension<T extends IndexableObj> = {
   deleteFromIndex: () => void ;
+  // It looks like Omit<T & Extension<T>, keyof Extension<T>> might just equal T.
+  // But that's not true if T intersects with Extension eg T includes a field called 'deleteFromIndex' or 'getTarget'
+  // We could probably guard against this using a conditional type... but let's not get too carried away
+  getTarget: () => Omit<T & Extension<T>, keyof Extension<T>>;
```

Comment:
I think we could just add 

```ts
  deleteFromIndex: never;
  getTarget: never;
```

to the definition of `IndexableObj` and then `getTarget: () => T;` should be valid

Reply:
Thanks, I think you're right! That's much easier than a conditional type.